src/components/views/rooms/RoomHeader.tsx

```diff
@@ -74,6 +76,7 @@ export default function RoomHeader({ room }: { room: Room }): JSX.Element {
     const memberCount = useRoomMemberCount(room, { throttleWait: 2500 });
 
     const { voiceCallDisabledReason, voiceCallClick, videoCallDisabledReason, videoCallClick } = useRoomCall(room);
+    const isVideoRoom = useFeatureEnabled("feature_video_rooms") && calcIsVideoRoom(room);
```

Comment:
this logic seems really rather magical. It doesn't appear to match the logic in `RoomView` that determines whether to show this as a video room or not. Can you explain to me why this is correct?

Reply:
> This is added to check whether the room is a video room or a normal chat room so that the chat icon is only visible in video rooms.

Yes I know, but I don't understand why is this the correct way to calculate it.

@robintown as the author of the code where this has been cargo-culted from (https://github.com/matrix-org/matrix-react-sdk/pull/9267/files#diff-137182f1b4995aeb920915d00e7e6c752dc38cb8ac5c6f62d74e54a3755dbd4bR316), can you help us understand if this is correct?

Reply:
Hi, this does look correct. In the case of the room header, the application should only acknowledge the existence of video rooms if the feature flag is enabled, and given that it is enabled, `calcIsVideoRoom` determines exactly that information.

I'm not sure what specific logic in RoomView you were looking at, but `calcIsVideoRoom` is used there as well (though imported instead under the name `isVideoRoom`). RoomView doesn't always check the feature flag though, because it is supposed to present a prompt to join the beta if it detects a video room. Maybe that's what was causing the confusion?

For that reason, and the fact that `isVideoRoom` may be called from a context in which React hooks such as `useFeatureFlag` can't be used, I preferred to keep the feature flag check separate and explicit like this.

Reply:
The code I was looking at in `RoomView` is here: https://github.com/matrix-org/matrix-react-sdk/blob/67452f50217778d7e20816554cef8b0680c0633b/src/components/structures/RoomView.tsx#L605-L610

 The main confusion for me is why we check the `feature_group_calls` setting and the `roomViewStore` in the `RoomView`, but not here. (I have no real idea what the different feature flags mean, and how `feature_group_calls`, `feature_video_rooms` and `feature_element_call_video_rooms` interact.) They just look like completely different logic.

I guess if you @robintown say it's fine then it's fine but I must say I find all this logic confusing and it would be nice if there were some comments to help people understand what the logic is actually doing. I mean, a naive reading would assume that a function called `RoomViewStore.isViewingCall`, or a function called `isVideoRoom`, would tell us all we need to know here.